packages/core/src/extensions/SideMenu/SideMenuPlugin.ts

```diff
@@ -268,30 +274,71 @@ export class SideMenuView<
     }
   };
 
-  /**
-   * If the event is outside the editor contents,
-   * we dispatch a fake event, so that we can still drop the content
-   * when dragging / dropping to the side of the editor
-   */
   onDrop = (event: DragEvent) => {
+    if (this.pmView.dragging === null) {
+      return;
+    }
+
     this.editor._tiptapEditor.commands.blur();
 
-    // ProseMirror doesn't remove the dragged content if it's dropped outside
-    // the editor (e.g. to other editors), so we need to do it manually. Since
-    // the dragged content is the same as the selected content, we can just
-    // delete the selection.
-    if (this.isDragOrigin && !this.pmView.dom.contains(event.target as Node)) {
-      this.pmView.dispatch(this.pmView.state.tr.deleteSelection());
+    if (this.sideMenuDetection === "editor") {
```

Comment:
why only run this on `this.sideMenuDetection === "editor"`?

Reply:
The PM default behaviour works fine when dragging and dropping blocks within the same editor, and the code here handles cases when blocks are being dragged and dropped between editors.

This assumes that `sideMenuDetection === "viewport"` only supports one editor at a time, and you need to use `sideMenuDetection === "editor"` to support multiple editors on the same page. That is currently true, but maybe in the future we will revisit it.